### PR TITLE
v0.21: ranking rules

### PR DESCRIPTION
**Pull request by dichotommy**
Removed wordsPosition, switched order of typo and words.

### CODE REVIEW

File: learn/core_concepts/relevancy.md [hunk 1]
```diff
@@ -16,28 +16,29 @@ The order in which they are applied has a significant impact on the search resul
...
-Results are sorted by **increasing number of typos**: find documents that match query terms with fewer typos first.
-
-**2. Words**
+**1. Words**
 Results are sorted by **decreasing number of matched query terms** in each matching document: find documents that contain more occurrences of the query terms first.
 
-::: warning
+::: note
+
+Be aware that the `words` rule works from right to left, so order of the query string does affect the sorting results.
```
Comment (dichotommy) [inline learn/core_concepts/relevancy.md:24]:
```suggestion
Be aware that the `words` rule works from right to left. Therefore, the order of search results is affected by the order of the query string.
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -150,7 +150,6 @@ update_ranking_rules_1: |-
         "words",
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:150]: I would keep the `words` before `typo` to discourage people swap them.

**Review by curquiza — CHANGES_REQUESTED** [review]

File: learn/core_concepts/relevancy.md [hunk 3]
```diff
@@ -16,28 +16,29 @@ The order in which they are applied has a significant impact on the search resul
...
-**2. Words**
+**1. Words**
 Results are sorted by **decreasing number of matched query terms** in each matching document: find documents that contain more occurrences of the query terms first.
 
-::: warning
+::: note
+
+Be aware that the `words` rule works from right to left. Therefore, the order of the query string impacts the order of results.
+
+For example, if someone were to search `batman dark knight`, then the `words` rule would rank documents containing all three terms first, documents containing only `batman` and `dark` second, and documents containing only `batman` third. All other documents would be ranked last.
```
Comment (Kerollmops) [inline learn/core_concepts/relevancy.md:26]: > All other documents would be ranked last.

Do you mean documents that contain only the word `batman` but with typos? I prefer that we remove this sentence. Only the documents that contain `batman` will be ranked last.

File: learn/core_concepts/relevancy.md [hunk 4]
```diff
@@ -16,28 +16,29 @@ The order in which they are applied has a significant impact on the search resul
...
 
-It is now mandatory that all query terms are present in the returned documents. This rule does not impact search results yet. <Badge text="soon" type="warn"/>
+In other words, results containing just `dark` and `knight`, or `batman` and `knight`, would be sorted last, equivalent to documents that contain none of the query terms.
```
Comment (Kerollmops) [inline learn/core_concepts/relevancy.md:28]: This cannot be possible, there is no way that the engine return documents that do not contain the first word as the engine tries to remove the words from the end of the query and always requires the documents to contain all of the query words (currently tested). You can remove this sentence :)

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: learn/core_concepts/relevancy.md (hunk 4 above)
Comment (dichotommy) [reply learn/core_concepts/relevancy.md:28]: Thanks for clarifying this @Kerollmops !

**Review by curquiza — APPROVED** [review]
🔥

**Review by Kerollmops — APPROVED** [review]
Nice work :)

**Pull request merged**